spot_wrapper/spot_arm.py

```diff
@@ -437,63 +437,73 @@ def gripper_angle_open(self, gripper_ang: float, ensure_power_on_and_stand: bool
 
         return True, "Opened gripper successfully"
 
-    def hand_pose(self, data) -> typing.Tuple[bool, str]:
+    def hand_pose(
+        self,
+        *,
+        x: float,
+        y: float,
+        z: float,
+        qx: float,
+        qy: float,
+        qz: float,
+        qw: float,
+        duration: float = 0.125,
+        ref_frame: str = "body",
+        ensure_power_on_and_stand: bool = True,
+    ) -> typing.Tuple[bool, str]:
         """
         Set the pose of the hand
 
         Args:
-            data:
+            x, y, z: float positions for the pose
+            qx, qy, qz, qw: float quaternion for the pose
+            duration: secodns for the trajectory
+            ref_frame: base frame for the pose. This needs to be something Spot knows about, ie "body" or "arm0.link_sh0"
+            ensure_power_on_and_stand: bool for whether or not to ensure Spot is standing/powered on before executing
 
         Returns:
             Boolean success, string message
         """
         try:
-            success, msg = self.ensure_arm_power_and_stand()
-            if not success:
-                self._logger.info(msg)
-                return False, msg
+            if ensure_power_on_and_stand:
+                success, msg = self.ensure_arm_power_and_stand()
+                if not success:
+                    self._logger.info(msg)
+                    return False, msg
             else:
-                pose_point = data.pose_point
-                # Move the arm to a spot in front of the robot given a pose for the gripper.
-                # Build a position to move the arm to (in meters, relative to the body frame origin.)
-                position = geometry_pb2.Vec3(
-                    x=pose_point.position.x,
-                    y=pose_point.position.y,
-                    z=pose_point.position.z,
-                )
+                powered_on = self._robot.is_powered_on()
+                if not powered_on:
+                    return False, "Robot not powered on and will not force power on"
 
-                # # Rotation as a quaternion.
-                rotation = geometry_pb2.Quaternion(
-                    w=pose_point.orientation.w,
-                    x=pose_point.orientation.x,
-                    y=pose_point.orientation.y,
-                    z=pose_point.orientation.z,
-                )
+            # Move the arm to a spot in front of the robot given a pose for the gripper.
+            # Build a position to move the arm to (in meters, relative to the body frame origin.)
+            position = geometry_pb2.Vec3(x=x, y=y, z=z)
 
-                seconds = data.duration
-                duration = seconds_to_duration(seconds)
+            # # Rotation as a quaternion.
```

Comment:
nit: remove double comment markers

```suggestion
            # Rotation as a quaternion.